app/controllers/staff/training_sessions_controller.rb

```diff
@@ -1,116 +1,63 @@
 class Staff::TrainingSessionsController < StaffAreaController
-  before_action :load_user, only: [:show, :edit, :update]
+  before_action :current_training_session, only: [:show, :edit, :update, :certify_trainees, :delete_training_session]
 
   layout 'staff_area'
 
   def index
-    if sort_params
-      if params[:p] == "signed_in_users"
-        if !params[:sort].present? && !params[:direction].present?
-          params[:sort] = "lab_sessions.sign_in_time"
-          params[:direction] = "desc"
-        end
-        @users_temp = LabSession.joins(:user).where("sign_out_time > ?", Time.now)
-        if params[:location].present?
-          @users_temp = @users_temp.joins("INNER JOIN pi_readers ON pi_mac_address = mac_address AND LOWER(pi_location) = LOWER('#{params[:location]}')")
-        end
-        @users_temp = @users_temp.order("#{params[:sort]} #{params[:direction]}").paginate(:page => params[:page], :per_page => 20)
-        @users = @users_temp.includes(:user).map{|session| session.user}
-        @total_pages = @users_temp.total_pages
-      elsif params[:p] == "new_users" || !params[:p].present?
-        if !params[:sort].present? && !params[:direction].present?
-          params[:sort] = "users.created_at"
-          params[:direction] = "desc"
-        end
-        @users = User.includes(:lab_sessions).order("#{params[:sort]} #{params[:direction]}").paginate(:page => params[:page], :per_page => 20)
-        @total_pages = @users.total_pages
-      end
-    else
-      redirect_to staff_index_path
-      flash[:alert] = "Invalid parameters!"
-    end
-  end
 
-  def sort_params
-    if (((["username","name","lab_sessions.sign_in_time","users.created_at"].include? params[:sort]) && (["desc","asc"].include? params[:direction])) || (!params[:sort].present? && !params[:direction].present?))
-      return true
-    end
   end
 
   def new
-    training_session = TrainingSession.new(training_session_params)
+    @new_training_session = TrainingSession.new
+  end
+
+  def create
+    @training_session = TrainingSession.new(training_session_params)
     if params['training_session_users'].present?
       params['training_session_users'].each do |user|
-        training_session.users << User.find(user)
+        @training_session.users << User.find(user)
       end
     end
-    redirect_to :back
-  end
-
-  def create
-    staff = current_user
-    training_session = TrainingSession.new(training_session_params)
-    if training_session.save
+    if @training_session.save
       flash[:notice] = "Training session created succesfully"
+      render json: { redirect_uri: "#{staff_training_session_path(@training_session.id)}" }
+    else
+      flash[:alert] = "Something went wrong. Please try again."
+      redirect_to :back
     end
-    redirect_to staff_training_sessions_url
   end
 
-  def rename_training_session
-    staff = current_user
-    training_session = TrainingSession.find_by(training_session_params)
+  def update
```

Comment:
You may be able to replace much of this with just `@current_training_session.update(training_session_params)` or something like that